clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -179,4 +185,17 @@ private static MethodHandle findConstructor(String className, MethodType methodT
         }
     }
 
+    static class RecyclingBufferSupplier implements KafkaLZ4BlockInputStream.BufferSupplier {
+
+        ByteBuffer theBuffer;
+
+        @Override
+        public ByteBuffer get(int size) {
+            if (theBuffer == null || theBuffer.capacity() < size) {
+                theBuffer = ByteBuffer.allocate(size);
+            }
+            theBuffer.limit(size);
+            return theBuffer;
```

Comment:
@xvrl Yeah, figured that was the case, but it makes the code a bit brittle to depend on that.